asreview/webapp/api/auth.py

```diff
@@ -0,0 +1,512 @@
+                result = (403, f"Unable to to confirm user! Reason: {str(e)}")
+
+    status, message = result
+    response = jsonify({"message": message})
+    return response, status
+
+
+@bp.route("/update_profile", methods=["POST"])
+@asreview_login_required
+def update_profile():
+    """Update user profile"""
+    user = User.query.get(current_user.id)
+    if user:
+        email = request.form.get("email", "").strip()
+        name = request.form.get("name", "").strip()
+        affiliation = request.form.get("affiliation", "").strip()
+        password = request.form.get("password", None)
+        public = bool(int(request.form.get("public", "1")))
+
+        try:
+            user = user.update_profile(email, name, affiliation, password, public)
+            DB.session.commit()
+            result = (200, "User profile updated")
+        except IntegrityError as e:
+            DB.session.rollback()
+            result = (500, f"Unable to update your profile! Reason: {str(e)}")
+        except SQLAlchemyError as e:
+            DB.session.rollback()
+            result = (500, f"Unable to update your profile! Reason: {str(e)}")
+
+    else:
+        result = (404, "No user found")
+
+    status, message = result
+    response = jsonify({"message": message})
+    return response, status
+
+
+@bp.route("/reset_password", methods=["POST"])
+def reset_password():
+    """Resests password of user"""
+    new_password = request.form.get("password", "").strip()
+    token = request.form.get("token", "").strip()
+    user_id = request.form.get("user_id", "0").strip()
+
+    user = User.query.get(user_id)
+    if not user:
+        result = (404, "User not found, try restarting the forgot-password procedure.")
+    elif not user.token_valid(token, max_hours=24):
+        result = (404, "Token is invalid, restart the forgot-password procedure.")
+    else:
+        try:
+            user = user.reset_password(new_password)
+            DB.session.commit()
+            result = (200, "Password updated")
+        except IntegrityError as e:
+            DB.session.rollback()
+            result = (500, f"Unable to reset your password! Reason: {str(e)}")
+        except SQLAlchemyError as e:
+            DB.session.rollback()
+            result = (500, f"Unable to reset your password! Reason: {str(e)}")
+
+    status, message = result
+    response = jsonify({"message": message})
+    return response, status
+
+
+@bp.route("/refresh", methods=["GET"])
+def refresh():
+    if current_user and isinstance(current_user, User):
+        logged_in = current_user.is_authenticated
+        name = current_user.get_name()
+        id = current_user.id
+    else:
+        logged_in = False
+        name = ""
+        id = None
+
+    result = {"logged_in": logged_in, "name": name, "id": id}
+
+    response = jsonify(result)
+    return response, 200
+
+
+@bp.route("/signout", methods=["DELETE"])
+@asreview_login_required
```

Comment:
I probably don't fully understand how this works, but this is the first time I see the `@asreview_login_required` decorator. Does this mean other routes like `/reset_password` don't need it?